### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by Omranic** [rinvex/repository#93]

**Comment by ebisbe** [comment]
Can I help in some way? @Omranic @ionut-tanasa

**Comment by Omranic** [comment]
@ebisbe Yes, yes, yes please 🙏  That would greatly help us release next version faster, we need to cover all functions with test suites, update missing parts in the docs and release it 😄  Appreciated in advance 👍

**Comment by ebisbe** [comment]
@Omranic ok, but thats a vague description of all the work. Where do I start?

**Comment by Omranic** [comment]
Honestly I still didn't scan for what's already implemented and what's missing, so if you can find your way inside the `tests` directory that would be easier and faster than waiting me analyze our current state 😅  @evsign & @ionut-tanasa who already worked thanksfuly on this before
